### DISCUSSION TITLE
Default Compiler: “/usr/local/texlive/2021/bin/universal-darwin/pdftex”

### DISCUSSION THREAD

**Discussion by YiCao5** [texstudio-org/texstudio#1583] [Q&A] [answered]
Hello, (please excuse me if this is not the right place to ask this question).

I have had no luck to get texstudio to work and it is really discouraging. I posted it here [https://tex.stackexchange.com/questions/592892/texstudio-could-not-start-default-compilerdefault-compiler-usr-local-texliv](https://tex.stackexchange.com/questions/592892/texstudio-could-not-start-default-compilerdefault-compiler-usr-local-texliv) but don't get much help so far. Can someone please help to have a look at it? Much appreciated!

**Comment by sunderme** [comment, marked as answer]
first of all, set "default compiler" back to default value (txs:///pdflatex) This setting is intended to switch between different compiler e.g. lualatex,pdflatex,xelatex.
The compiler in the system is adapted with the commands options.

**Reply by YiCao5** [reply]
Thank you @sunderme ! I reset the default compiler and now get a new error "Process exited with error(s)", as in the screenshot. Where can I see the detailed error message?

![Screen Shot 2021-04-15 at 3 23 57 PM](https://user-images.githubusercontent.com/9289589/114926987-fa358e80-9dfe-11eb-92a4-2cd1bb30d2f9.png)

**Reply by sunderme** [reply]
Please execute help/check latex installation
Can you start pdflatex from a terimnal ?

**Reply by YiCao5** [reply]
**help/check latex installation**

dpi: 72
which pdflatex: /Library/TeX/texbin/pdflatex

PDFLATEX: "/Library/TeX/texbin/pdflatex" -version
pdfTeX 3.141592653-2.6-1.40.22 (TeX Live 2021)

Environment variables:
HOME=/Users/me
LOGNAME=me
PATH=/Users/me/opt/anaconda3/bin:/Users/me/opt/anaconda3/condabin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/texlive/2020:/usr/local/texlive/2021:/Library/TeX/texbin:/usr/local/texlive/2021/bin/universal-darwin
SHELL=/bin/bash

TeXstudio:
Path        : /Applications/texstudio.app/Contents/MacOS/texstudio
Program call: /Applications/texstudio.app/Contents/MacOS/texstudio
Setting file: /Users/me/.config/texstudio/texstudio.ini

Command configuration in TeXstudio:

    PdfLaTeX (pdflatex) (r): "/Library/TeX/texbin/pdflatex" -synctex=1 -interaction=nonstopmode %.tex
    LaTeX (latex) (r): "/Library/TeX/texbin/latex" -src -interaction=nonstopmode %.tex


Additional Search Paths:
    Command: /usr/local/texlive/2020:/usr/local/texlive/2021:/Library/TeX/texbin:/usr/local/texlive/2021/bin/universal-darwin
    Log: 
    Pdf: 

**terminal pdflatex**
$ /Library/TeX/texbin/pdflatex
This is pdfTeX, Version 3.141592653-2.6-1.40.22 (TeX Live 2021) (preloaded format=pdflatex)
 restricted \write18 enabled.
**
Please type the name of your input file.
**test.tex
entering extended mode
! I can't find file `test.tex'.
<*> test.tex
            
(Press Enter to retry, or Control-D to exit)

**Reply by sunderme** [reply]
it looks that you have two versions of texlive on your system
and you have changed "additional command path set"

try to compile your test document from the command line. If that works, maybe goind back to default settings for txs ?

**Reply by YiCao5** [reply]
@sunderme Thank you!
command line compiling works as below. Then what do you mean by going back to default settings for txs? what is the default settings? 

$ more Untitled.tex
\documentclass{article}         % use "amsart" instead of 

\begin{document}
        
        hello Latex
        
\end{document}  
$ pdflatex
This is pdfTeX, Version 3.141592653-2.6-1.40.22 (TeX Live 2021) (preloaded format=pdflatex)
 restricted \write18 enabled.
**
Please type the name of your input file.
**Untitled.tex
entering extended mode
(./Untitled.tex
LaTeX2e <2020-10-01> patch level 4
L3 programming layer <2021-02-18>
(/usr/local/texlive/2021/texmf-dist/tex/latex/base/article.cls
Document Class: article 2020/04/10 v1.4m Standard LaTeX document class
(/usr/local/texlive/2021/texmf-dist/tex/latex/base/size10.clo))
(/usr/local/texlive/2021/texmf-dist/tex/latex/l3backend/l3backend-pdftex.def)
(./Untitled.aux) [1{/usr/local/texlive/2021/texmf-var/fonts/map/pdftex/updmap/p
dftex.map}] (./Untitled.aux) )</usr/local/texlive/2021/texmf-dist/fonts/type1/p
ublic/amsfonts/cm/cmr10.pfb>
Output written on Untitled.pdf (1 page, 11222 bytes).
Transcript written on Untitled.log.

**Reply by sunderme** [reply]
usally you would use the command: pdflatex **filename**

Anyway, close txs, remove ~/.config/texstudio/texstudio and start txs again

**Reply by YiCao5** [reply]
you mean remove the **`texstudio.ini`** file as below?
$ pwd
~/.config/texstudio
$ l
total 320
-rw-r--r--  1 Users  17516 16 Apr 11:41 texstudio.ini

**Reply by sunderme** [reply]
yes

**Reply by YiCao5** [reply]
txs still not working - the outputs is as the screenshot below. However, pdflatex **finename** works and generate the pdf file properly. 

![Screen Shot 2021-04-16 at 2 47 33 PM](https://user-images.githubusercontent.com/9289589/115070709-4ba75180-9ec3-11eb-9af9-f3840fd80740.png)

![Screen Shot 2021-04-16 at 2 52 26 PM](https://user-images.githubusercontent.com/9289589/115070794-67125c80-9ec3-11eb-8f0c-07c1d935f92e.png)
![Screen Shot 2021-04-16 at 2 52 55 PM](https://user-images.githubusercontent.com/9289589/115070817-70032e00-9ec3-11eb-904c-417f8f29eba2.png)

**Reply by sunderme** [reply]
The relevant info is a) output messages of pdflatex and b) log of pdflatex run

**Reply by YiCao5** [reply]
**a) output message**

Process started: "/Library/TeX/texbin/pdflatex" -synctex=1 -interaction=nonstopmode "Untitled".tex

/Library/TeX/texbin/pdflatex: getcwd: No such file or directory
Process exited with error(s)

Process started: "/Library/TeX/texbin/pdflatex" -synctex=1 -interaction=nonstopmode "Untitled".tex

/Library/TeX/texbin/pdflatex: getcwd: No such file or directory
Process exited with error(s)

**b) log of pdflatex**
This is pdfTeX, Version 3.141592653-2.6-1.40.22 (TeX Live 2021) (preloaded format=pdflatex 2021.4.6)  16 APR 2021 16:01
entering extended mode
 restricted \write18 enabled.
 %&-line parsing enabled.
**Untitled.tex
(./Untitled.tex
LaTeX2e <2020-10-01> patch level 4
L3 programming layer <2021-02-18>
(/usr/local/texlive/2021/texmf-dist/tex/latex/base/article.cls
Document Class: article 2020/04/10 v1.4m Standard LaTeX document class
(/usr/local/texlive/2021/texmf-dist/tex/latex/base/size10.clo
File: size10.clo 2020/04/10 v1.4m Standard LaTeX file (size option)
)
\c@part=\count179
\c@section=\count180
\c@subsection=\count181
\c@subsubsection=\count182
\c@paragraph=\count183
\c@subparagraph=\count184
\c@figure=\count185
\c@table=\count186
\abovecaptionskip=\skip47
\belowcaptionskip=\skip48
\bibindent=\dimen138
)
(/usr/local/texlive/2021/texmf-dist/tex/latex/l3backend/l3backend-pdftex.def
File: l3backend-pdftex.def 2021-03-18 L3 backend support: PDF output (pdfTeX)
\l__color_backend_stack_int=\count187
\l__pdf_internal_box=\box47
)
(./Untitled.aux

**Reply by sunderme** [reply]
https://tex.stackexchange.com/questions/511617/error-with-pdftex-after-updating-to-mac-os-catalina ?

**Reply by YiCao5** [reply]
Thank you so much @sunderme !!! changing the full access fixes the issue. txs works now!
